### DISCUSSION TITLE
Multiple 1-to-1 relationships on a single model

### DISCUSSION THREAD

**Discussion by jogibs** [prisma/prisma#12375] [Q&A] [answered]
I have a problem with Prisma, which I can't really figure out. I have the following schema:

```
model Order {
  id Int @id @default(autoincrement())
  ...
  tradeId Int
  trade Trade @relation(name: "Order_Trade", fields: [tradeId], references: [id])
}

model Trade {
  id Int @id @default(autoincrement())
  ...
  entryOrderId Int
  entryOrder Order @relation(name: "Trade_EntryOrder")
  exitOrderId Int
  exitOrder Order @relation(name: "Trade_ExitOrder")
}
```

Now, I'm pretty sure the Order part should be fine, but I simply can't figure out bind two models of the same type to it, in an 1-to-1 relationship, because in the code later, I want to get the trade by just getting the order and then including the trade, instead of creating separate tradeEntryOrders and tradeExitOrders (had that one initially) and looping through them all, to find the trade it actually belongs to.

Any help/idea would really be appreciated.

**Comment by nurul3101** [comment, marked as answer]
Hey @jogibs 👋 

Can you try out this relations and see if this solves it for you?
```prisma
model Order {
  id               Int     @id @default(autoincrement())
  tradeId          Int
  trade            Trade   @relation(name: "Order_Trade", fields: [tradeId], references: [id])
  entryOrderTrades Trade[] @relation("Trade_EntryOrder")
  exitOrderTrades  Trade[] @relation("Trade_ExitOrder")
}

model Trade {
  id           Int     @id @default(autoincrement())
  entryOrderId Int
  entryOrder   Order   @relation(name: "Trade_EntryOrder", fields: [entryOrderId], references: [id])
  exitOrderId  Int
  exitOrder    Order   @relation(name: "Trade_ExitOrder", fields: [exitOrderId], references: [id])
  order        Order[] @relation("Order_Trade")
}
```
I have taken reference from this [Disambiguating Relation Fields](https://www.prisma.io/docs/concepts/components/prisma-schema/relations#disambiguating-relations) Guide.

**Reply by jogibs** [reply]
Ah brilliant, that works! Thank you very much!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
